cache.go

```diff
@@ -118,12 +119,35 @@ func (c *Client[T]) performContinuousEvictions() {
 
 // getShard returns the shard that should be used for the specified key.
 func (c *Client[T]) getShard(key string) *shard[T] {
-	hash := xxhash.Sum64String(key)
-	shardIndex := hash % uint64(len(c.shards))
-	c.reportShardIndex(int(shardIndex))
+	shardIndex := c.getShardIndex(key)
 	return c.shards[shardIndex]
 }
 
+func (c *Client[T]) getShardIndex(key string) int {
+	// They key could be an alias, so let's first search for the shardIndex by alias
+	shardIndex := c.findShardIndexByAlias(key)
+	if shardIndex == -1 {
+		// If the key is not a known alias, it may be a primary key.
+		// We can use the hash to determine the shard index.
+		hash := xxhash.Sum64String(key)
+		shardIndex = int(hash % uint64(len(c.shards)))
+	}
+	c.reportShardIndex(shardIndex)
+	return shardIndex
+}
+
+func (c *Client[T]) findShardIndexByAlias(key string) int {
+	for shardIndex, shard := range c.shards {
```

Comment:
This loop RLock's every shard one at a time while it searches for alias conflicts. I don't think this will be a performance issue, but please let me know what you think. According to my understanding of golang's concurrency model, this will thread nicely.